### PR TITLE
Reactivate codecov

### LINKED ISSUE #13: Reactivate codecov in CI
Codecov had issues from the very beginning.

```
Request failed [404]. Retrying in 1 seconds...
Request failed [404]. Retrying in 4 seconds...
$detail
[1] "Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue."
```

https://github.com/Appsilon/box.linters/actions/runs/8826917662/job/24233363150

### PR DESCRIPTION
**Pull request by radbasa**
Closes #13 

## Description
* Activate test coverage stage in CI
* Added codecov badge to readme
## Definition of Done
- [ ] The change is thoroughly documented.
- [ ] The CI passes (`R CMD check`, linter, unit tests, spelling).
- [ ] Any generated files have been updated (e.g. `.Rd` files with `roxygen2::roxygenise()`)

### CODE REVIEW

**Comment by radbasa** [comment]
Code cov still failing

File: README.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 <!-- badges: start -->
 [![CRAN status](https://www.r-pkg.org/badges/version/box.linters)](https://cran.r-project.org/package=box.linters)
 [![R-CMD-check](https://github.com/Appsilon/box.linters/workflows/CI/badge.svg)](https://github.com/Appsilon/box.linters/actions/workflows/ci.yml)
+![Codecov test coverage](https://codecov.io/gh/Appsilon/box.linters/branch/main/graph/badge.svg)
```
Comment (jakubnowicki) [inline README.md:8]:
```suggestion
[![codecov](https://codecov.io/gh/Appsilon/box.linters/graph/badge.svg?token=GW0DLN6ERF)](https://codecov.io/gh/Appsilon/box.linters)
```

**Review by jakubnowicki — APPROVED** [review]
Works after adding the env variable.

File: README.md (hunk 1 above)
Comment (radbasa) [reply README.md:8]: I suppose this suggested change is now not necessary.

**Pull request merged**